test/test_base_service.py

```diff
@@ -432,8 +432,8 @@ def test_request_success_nonjson():
     service = AnyServiceV1('2018-11-20', authenticator=NoAuthAuthenticator())
     prepped = service.prepare_request('GET', url='')
     detailed_response = service.send(prepped)
-    # It's odd that we have to call ".text" to get the string value
-    # (see issue 3557)
+    assert repr(detailed_response.get_result()) == '<h1>Hola, amigo!</h1>'
+    assert str(detailed_response.get_result()) == '<h1>Hola, amigo!</h1>'
```

Comment:
Still not perfect... `repr` or `str` has to be called to get the string value. It's sad, because in the test output it almost looks good
```
AssertionError: assert <h1>Hola, amigo!</h1> == '<h1>Hola, amigo!</h1>'
```
but the first one is actually an instance of the `TextResult` object and that's just the representation. That means we can't
```
'Result:' + response.get_result()
```
but
`'Result:' + str(response.get_result())` or `'Result: {0}'.format(response.get_result())`